src/main/java/dev/openfeature/sdk/ImmutableStructure.java

```diff
@@ -50,7 +50,7 @@ public Set<String> keySet() {
     @Override
     public Value getValue(String key) {
         Value value = this.attributes.get(key);
-        return value.clone();
+        return value != null ? value.clone() : null;
```

Comment:
Thanks for finding/fixing/testing this. I don't think we have a similar issue in `MutableStructure` (the older implementation) because we don't clone the retrieved value.